llvm/lib/TargetParser/Host.cpp

```diff
@@ -1495,14 +1540,30 @@ StringRef sys::getHostCPUName() {
   case CPUFAMILY_ARM_LIGHTNING_THUNDER:
     return "apple-a13";
   case CPUFAMILY_ARM_FIRESTORM_ICESTORM:
-    return "apple-m1";
+    return "apple-m1"; // A14 / M1
   case CPUFAMILY_ARM_BLIZZARD_AVALANCHE:
-    return "apple-m2";
+    return "apple-m2"; // A15 / M2
   case CPUFAMILY_ARM_EVEREST_SAWTOOTH:
-    return "apple-m3";
+    return "apple-m3"; // A16
+  case CPUFAMILY_ARM_IBIZA:
+    return "apple-m3"; // M3
+  case CPUFAMILY_ARM_PALMA:
+    return "apple-m3"; // M3 Max
+  case CPUFAMILY_ARM_COLL:
+    return "apple-m3"; // A17
+  case CPUFAMILY_ARM_LOBOS:
+    return "apple-m3"; // M3 Pro
+  case CPUFAMILY_ARM_DONAN:
+    return "apple-m4"; // M4
+  case CPUFAMILY_ARM_BRAVA:
+    return "apple-m4"; // M4 Max
+  case CPUFAMILY_ARM_TAHITI:
+    return "apple-m4"; // A18 Pro
+  case CPUFAMILY_ARM_TUPAI:
+    return "apple-m4"; // A18
```

Comment:
Mind combining these with fallthroughs?

Reply:
I kinda liked the explicitness, but sure, have done so now.